src/udev/udevadm-monitor.c

```diff
@@ -41,9 +41,9 @@ static void print_device(struct udev_device *device, const char *source, int pro
         struct timespec ts;
 
         assert_se(clock_gettime(CLOCK_MONOTONIC, &ts) == 0);
-        printf("%-6s[%"PRI_TIME".%06ld] %-8s %s (%s)\n",
+        printf("%-6s[%"PRI_TIME".%06lu] %-8s %s (%s)\n",
                source,
-               ts.tv_sec, ts.tv_nsec/1000,
+               ts.tv_sec, (unsigned long)ts.tv_nsec/1000,
```

Comment:
I'd prefer `/ 1000LU` instead of a cast here too. The advantage is that if we get the types wrong (and tv_nsec is longer than unsigned long), we will not get a narrowing cast.